Dalamud/Game/Gui/Nameplates/NameplateGui.cs

```diff
@@ -0,0 +1,178 @@
+        return this.GetNameplateGameObject<T>(nameplateObject.Pointer);
+    }
+
+    /// <inheritdoc/>
+    public T? GetNameplateGameObject<T>(IntPtr nameplateObjectPtr) where T : GameObject
+    {
+        // Get the nameplate object array
+        var nameplateAddonPtr = this.gameGui.GetAddonByName("NamePlate", 1);
+        var nameplateObjectArrayPtrPtr = nameplateAddonPtr + Marshal.OffsetOf(typeof(AddonNamePlate), nameof(AddonNamePlate.NamePlateObjectArray)).ToInt32();
+        var nameplateObjectArrayPtr = Marshal.ReadIntPtr(nameplateObjectArrayPtrPtr);
+
+        if (nameplateObjectArrayPtr == IntPtr.Zero)
+            return null;
+
+        // Determine the index of the nameplate object within the nameplate object array
+        var namePlateObjectSize = Marshal.SizeOf(typeof(AddonNamePlate.NamePlateObject));
+        var namePlateObjectPtr0 = nameplateObjectArrayPtr + namePlateObjectSize * 0;
+        var namePlateIndex = (nameplateObjectPtr.ToInt64() - namePlateObjectPtr0.ToInt64()) / namePlateObjectSize;
+        
+        if (namePlateIndex < 0 || namePlateIndex >= AddonNamePlate.NumNamePlateObjects)
+            return null;
+
+        // Get the nameplate info array
+        var nameplateInfoArrayPtr = IntPtr.Zero;
+        unsafe
+        {
+            var framework = FFXIVClientStructs.FFXIV.Client.System.Framework.Framework.Instance();
+            nameplateInfoArrayPtr = new IntPtr(&framework->GetUiModule()->GetRaptureAtkModule()->NamePlateInfoArray);
+        }
+
+        // Get the nameplate info for the nameplate object
+        var namePlateInfoPtr = new IntPtr(nameplateInfoArrayPtr.ToInt64() + Marshal.SizeOf(typeof(RaptureAtkModule.NamePlateInfo)) * namePlateIndex);
+        var namePlateInfo = Marshal.PtrToStructure<RaptureAtkModule.NamePlateInfo>(namePlateInfoPtr);
+
+        // Return the object for its object id
+        var objectId = namePlateInfo.ObjectID.ObjectID;
+        return this.objectTable.SearchById(objectId) as T;
+    }
+
+    private IntPtr HandleSetPlayerNameplateDetour(IntPtr playerNameplateObjectPtr, bool isTitleAboveName, bool isTitleVisible, IntPtr titlePtr, IntPtr namePtr, IntPtr freeCompanyPtr, IntPtr prefixPtr, int iconId)
+    {
+
+        if (this.OnNameplateUpdate != null)
+        {
+            // Create NamePlateObject if possible
+            var namePlateObj = new NameplateObject(playerNameplateObjectPtr);
+
+            // Create new event
+            var nameplateInfo = new NameplateInfo
+            {
+                Title = MemoryHelper.ReadSeStringNullTerminated(titlePtr),
+                Name = MemoryHelper.ReadSeStringNullTerminated(namePtr),
+                FreeCompany = MemoryHelper.ReadSeStringNullTerminated(freeCompanyPtr),
+                Prefix = MemoryHelper.ReadSeStringNullTerminated(prefixPtr),
+                IsTitleAboveName = isTitleAboveName,
+                IsTitleVisible = isTitleVisible,
+                IconID = (StatusIcons)iconId,
+            };
+
```

Comment:
Is it necessary to parse this data every time? This is a lot of SeString parsing that is happening when it might not even be required. Might be better for NameplateInfo to store a address internally and offer this parsing as a get-only property.

Reply:
True. Moved that to `NameplateNode.Text`. It will be parsed when the getter is called the first time.